src/app.py

```diff
@@ -54,6 +59,15 @@ def handle_github_webhook() -> tuple[str, int]:
     Returns:
         A tuple containing an empty string and 200 status code.
     """
+    if secret := app.config.get("WEBHOOK_SECRET"):
+        if not (signature := request.headers.get(WEBHOOK_SIGNATURE_HEADER)):
+            return "X-Hub-signature-256 header is missing!", 403
+
+        if not verify_signature(
+            payload=request.data, secret_token=secret, signature_header=signature
+        ):
+            return "Signature validation failed!", 403
```

Comment:
Minor: Should both cases be logged at least as debug?

Reply:
Updated (note that application logs are not yet configured except in dev mode, but will be once the app gets charmed, https://github.com/canonical/github-runner-webhook-router/pull/2#issuecomment-1991708258)